apps/opik-backend/src/main/resources/liquibase/db-app-analytics/migrations/000009_add_duration_columns.sql

```diff
@@ -0,0 +1,19 @@
+--liquibase formatted sql
+--changeset idoberko2:add_duration_columns
+
+ALTER TABLE ${ANALYTICS_DB_DATABASE_NAME}.spans
+    ADD COLUMN IF NOT EXISTS duration_millis Nullable(Float64) MATERIALIZED
+        if(end_time IS NOT NULL AND start_time IS NOT NULL
+            AND notEquals(start_time, toDateTime64('1970-01-01 00:00:00.000', 9)),
+        (dateDiff('microsecond', start_time, end_time) / 1000.0),
+        NULL);
+
+ALTER TABLE ${ANALYTICS_DB_DATABASE_NAME}.traces
+    ADD COLUMN IF NOT EXISTS duration_millis Nullable(Float64) MATERIALIZED
+        if(end_time IS NOT NULL AND start_time IS NOT NULL
+            AND notEquals(start_time, toDateTime64('1970-01-01 00:00:00.000', 9)),
+        (dateDiff('microsecond', start_time, end_time) / 1000.0),
+        NULL);
```

Comment:
Forgot to mention, that my initial comment was miss-understood. 

I mentioned that the calculation had to be moved from the backend service side to the ClickHouse query time, not necessarily that we had to add a calculated field.

This works fine from the functional perspective, but this migration might cause issues when deploying to an environment where spans or traces tables are big, as the new field would need to be calculated for all rows.

We're still on time to re-consider this.

Reply:
Do you believe switching to a virtual column is a good alternative?

Reply:
Addressed in #916